sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/patch/PatchOperationCore.java

```diff
@@ -13,27 +13,43 @@
 public final class PatchOperationCore<T> extends PatchOperation {
 
     private final String path;
+    private final String from;
     private final T resource;
 
     /**
      * Initializes a new instance of the {@link PatchOperationCore} class.
      *
      * @param operationType Specifies the type of Update operation
      * @param path          Specifies the path to target location.
-     * @param value         Specifies the value to be used
+     * @param value         Specifies the value to be used. In case of move operation it will be a string in the format
+     *                      of a path
      */
     public PatchOperationCore(PatchOperationType operationType, String path, T value) {
         super(operationType);
 
         checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
-        this.path = path;
-        this.resource = value;
+
+        if(operationType == PatchOperationType.MOVE) {
+            checkArgument(StringUtils.isNotEmpty((String) value), "path empty %s", (String) value);
```

Comment:
This can throw `ClassCastException` which might get thrown to end user. We should check the value type and then validate it. If it is not `String` type, we should simply throw `IllegalArgumentException`